.github/workflows/periodic-merge.yml

```diff
@@ -0,0 +1,49 @@
+name: "Periodic Merges"
+
+on:
+  schedule:
+    # * is a special character in YAML so you have to quote this string
+    # Merge every 6 hours
+    - cron:  '0 */6 * * *'
+
+jobs:
+  periodic-merge:
+    if: github.repository_owner == 'NixOS'
+    runs-on: ubuntu-latest
+    strategy:
+      # don't fail fast, so that all pairs are tried
+      fail-fast: false
+      # certain branches need to be merged in order, like master->staging-next->staging
+      # and disabling parallelism ensures the order of the pairs below.
+      max-parallel: 1
+      matrix:
+        pairs:
+          - from: master
+            into: staging-next
+          - from: staging-next
+            into: staging
+          - from: release-21.05
+            into: staging-21.05-next
+          - from: staging-21.05-next
+            into: staging-21.05
+          - from: master
+            into: haskell-updates
+    name: ${{ matrix.pairs.from }} → ${{ matrix.pairs.into }}
+    steps:
+      - uses: actions/checkout@v2
```

Comment:
Does this mean we do 3 checkouts now? Those is the slowest step and usually takes a few minutes.

Reply:
Yes, that might be the case. The question is, does that cause any problems? This job runs every 6h and I don‘t think anyone cares how fast it is, do they?

Reply:
Also just for the record, as is, I think this are 5 checkouts …

Reply:
> Yes, that might be the case. The question is, does that cause any problems?

I don't think it can cause merge conflicts or so.